### DISCUSSION TITLE
classes/class_animatablebody2d

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#203] [User-contributed notes]
# classes/class_animatablebody2d

Inherits: StaticBody2D< PhysicsBody2D< CollisionObject2D< Node2D< CanvasItem< Node< Object A 2D physics body that can't be moved by external forces. When moved manually, it affects other bodies in ...

https://docs.godotengine.org/en/4.3/classes/class_animatablebody2d.html

**Comment by DCON-b** [comment]
If animating using AnimationPlayer, animate a RemoteTransform2D and have the AnimatableBody2D follow this, not the AnimatableBody2D directly. Enabling sync_to_physics and animating the AnimatableBody2D directly will cause the AnimatableBody2D not to move.

**Reply by AmirMdv** [reply]
Adding to this, if you are only using Script to move AnimatableBody2D you need to disable sync_to_physics or it won't collide with other bodies at all. Took me a whole day to figure it out.

**Reply by Calinou** [reply]
For future reference, this a known issue:

- https://github.com/godotengine/godot/issues/67257

**Reply by dginovker** [reply]
Solution seems to be `sync_to_physics = false`

**Comment by miketvo** [comment]
As mentioned by @NancokPS2 in the documentation page for StaticBody2D, it seems that this would be a better choice comparing to StaticBody when you need a PhysicsBody that moves manually:

>> Note: at least since 4.2 (possibly since the introduction of AnimatableBody) Areas do NOT detect moving StaticBodies.
This is not stated anywhere.

This should probably be stated on both documentation pages.

**Comment by DocMine** [comment]
```
extends AnimatableBody2D

func _physics_process(delta: float) -> void:
	
	if Input.is_action_pressed("ui_accept"):
		global_position.x = global_position.x + 50
		print(global_position.x)
	global_position.x -= 1
```
这段代码只会让我的节点向左运动，但在staticbody2d上工作正常；我认为是  sync_to_physics = true  导致了这一点。
